cloud/services/compute/firewalls.go

```diff
@@ -73,6 +73,11 @@ func (s *Service) DeleteFirewalls() error {
 }
 
 func (s *Service) getFirewallSpecs() []*compute.Firewall {
+	backendPort := int64(APIServerLoadBalancerBackendPort)
+	if s.scope.LoadBalancerBackendPort() != 0 {
+		backendPort = s.scope.LoadBalancerBackendPort()
+	}
```

Comment:
It seems like the default logic is duplicated here as well, let's make it happen in one place